pkg/minishift/profile/profile.go

```diff
@@ -0,0 +1,123 @@
+/*
+Copyright (C) 2017 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package profile
+
+import (
+	"fmt"
+
+	"github.com/minishift/minishift/pkg/minikube/constants"
+	"github.com/minishift/minishift/pkg/minishift/config"
+)
+
+//Returns the machine name after taking in to account if --profile or
+//profile information available in allinstance config
+func GetProfileName() string {
```

Comment:
If you don't want to pass 'config.AllInstancesConfig.Profiles' all the time, create a struct ProfileManager, add a constructor which takes _config.AllInstancesConfig.Profiles_ and keeps it as an instance member. GetProfileName become then members of this struct.